h2o-automl/src/main/java/ai/h2o/automl/ModelingStep.java

```diff
@@ -322,6 +324,8 @@ protected Work getAllocatedWork() {
             AutoMLBuildSpec buildSpec = aml().getBuildSpec();
             RandomDiscreteValueSearchCriteria searchCriteria =
                     (RandomDiscreteValueSearchCriteria) buildSpec.build_control.stopping_criteria.getSearchCriteria().clone();
```

Comment:
it seems like you want to "deep clone", function `set_stopping_criteria` seems unnecessary in the API IMHO

Reply:
yes, that was my original intention, but unfortunately `clone` is final in `Iced` class, so I have to create an ad-hoc `deepClone` method only for `HyperSpaceSearchCriteria` which I don't find completely elegant either... anyway it's still probably preferable to this double clone, so switching to a deepClone impl. and removing the then unnecessary setter.